__tests__/ship.test.js

```diff
@@ -0,0 +1,100 @@
+/* globals describe it expect */
+const Ship = require('../src/ship');
+const Port = require('../src/port');
+const Itinerary = require('../src/itinerary');
+
+describe('Ship constructor', () => {
+  describe('Ships with ports and an itinerary', () => {
+    let ship;
+    let dover;
+    let calais;
+    let itinerary;
+    // let port;
+
+    beforeEach(() => {
+      dover = {
+        addShip: jest.fn(),
+        removeShip: jest.fn(),
+        name: 'Dover',
+        ships: []
+      };
+
+      calais = {
+        addShip: jest.fn(),
+        removeShip: jest.fn(),
+        name: 'Calais',
+        ships: []
+      };
+      port = new Port('Dover'); 
+      // dover = new Port('Dover'); replaced with spies above.
+      // calais = new Port('Calais'); replaced with spies above.
+      itinerary = new Itinerary ([dover, calais]);
+      ship = new Ship(itinerary);
+    });
+
+    it('Ship can be instantiated', () => {
+      const port = new Port('Dover');
+      const itinerary = new Itinerary([port]);
+      const ship = new Ship(itinerary);
+
```

Comment:
You can remove all this setup since you are already doing it in the beforeEach block, theres no need to create these variables again